gems/aws-sdk-s3/lib/aws-sdk-s3/multipart_file_uploader.rb

```diff
@@ -187,6 +196,23 @@ def to_a
         end
 
       end
+
+      # @api private
+      class MultipartProgress
+        def initialize(parts, progress_callback)
+          parts = parts.to_a
```

Comment:
Check to see if this is done anywhere else, and if so, it maybe makes sense to implement size and map.

Reply:
`PartList#to_a` is also used when returning the completed parts ordered by `:part_number` which wouldn't be solved by adding `map` - instead, I'm adding a specific method for `part_sizes` as I don't think we want to expose map and I do prefer not using the `to_a` method here.